internal/tools/bigquery/bigqueryexecutesql/bigqueryexecutesql.go

```diff
@@ -85,7 +88,29 @@ func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error)
 		return nil, fmt.Errorf("invalid source for %q tool: source kind must be one of %q", kind, compatibleSources)
 	}
 
-	sqlParameter := tools.NewStringParameter("sql", "The sql to execute.")
+	sqlDescription := "The sql to execute."
+	allowedDatasets := s.BigQueryAllowedDatasets()
+	if len(allowedDatasets) > 0 {
+		datasetIDs := []string{}
+		for _, ds := range allowedDatasets {
+			datasetIDs = append(datasetIDs, fmt.Sprintf("`%s`", ds))
+		}
+
+		if len(datasetIDs) == 1 {
+			datasetFQN := allowedDatasets[0]
+			parts := strings.Split(datasetFQN, ".")
+			datasetID := datasetFQN
+			if len(parts) == 2 {
+				datasetID = parts[1]
+			}
```

Comment:
```suggestion
			parts := strings.Split(allowedDatasets[0], ".")
			if len(parts) <2 {
				return nil, fmt.Errorf("expected split to have 2 parts: %s", allowedDatasets[0])
			}
			datasetID := parts[1]
```

source seems to have made sure that the allowedDataset list consist of the full ID `project.dataset` format.